pinot-core/src/main/java/org/apache/pinot/core/operator/query/NonScanBasedAggregationOperator.java

```diff
@@ -122,8 +131,26 @@ protected IntermediateResultsBlock getNextBlock() {
     return new IntermediateResultsBlock(_aggregationFunctions, aggregationResults, false);
   }
 
-  private double toDouble(Comparable value) {
-    if (value instanceof Number) {
+  private static Double getMinValue(DataSource dataSource) {
+    Dictionary dictionary = dataSource.getDictionary();
+    if (dictionary != null) {
+      return toDouble(dictionary.getMinVal());
+    }
+    return toDouble(dataSource.getDataSourceMetadata().getMinValue());
```

Comment:
oh i meant can we create an `ImmutableList<Dictionary> dataSourceDictionaries` in constructor without extracting it on each block / agg function since the list of dataSource is final. 

But yeah agree it is not per row so perf difference should be minimal.

Reply:
It's only going to be called once per query in this context. There could be many aggregation functions but I'm tempted to keep things the way they are for the sake of simplicity.